src/constructs/numbers.md

```diff
@@ -2,6 +2,120 @@
 title: Numbers
 ---
 
-## On the Subject of Numbers
+Grain has a unified number type, called `Number`, which contains integer (whole) numbers, floating-point (decimal) numbers, and rational (fractional) numbers. Grain also has separate integer and float types for programs which require more precise typechecking.
 
-This documentation hasn't yet been migrated here (or written, to be honest). In the meanwhile, please check out the [Grain compiler README](https://github.com/grain-lang/grain/blob/master/README.md) for more information. Consider contributing by fleshing out this page yourself!
\ No newline at end of file
+```grain
+42 // an integer
+```
+
+```grain
+1.23 // a floating-point number
+```
+
+```grain
+1/3 // a rational number
+```
+
+## Number Syntax
+
+### Integers
+
+To make an integer more readable, underscores can be used throughout the number as a visual separator. The underscores do not affect the value of the number.
+
+```grain
+let oneBillion = 1_000_000_000
+```
+
+In addition to decimal numbers, integers can be written in hexadecimal, octal, and binary:
+
+```grain
+// hexadecimal
+let fourtyTwo = 0x2A // or 0x2a
+```
+
+```grain
+// octal
+let fourtyTwo = 0o52
+```
+
+```grain
+// binary
+let fourtyTwo = 0b101010
+```
+
+Of course, underscores can also be used with these other formats:
+
+```grain
+let dec0de = 0b1101_1110_1100_0000_1101_1110
```

Comment:
```suggestion
let decode = 0b1101_1110_1100_0000_1101_1110
```

Why a zero in the name?

Reply:
That's the binary sequence for the hexadecimal value `0xdec0de`. Could use something else I guess, but there aren't a ton of good sequences that are also words :/

Reply:
So punny 😛 I can see us getting "typo" PRs about it so maybe add a comment?

Reply:
Yeah, comment sounds good.